src/drive/web/modules/navigation/AppRoute.jsx

```diff
@@ -30,24 +30,27 @@ const AppRoute = (
         <Route path="uploadfrommobile" component={UploadFromMobile} />
       )}
       <Redirect from="/files/:folderId" to="/folder/:folderId" />
-      {flag('drive.client-migration.enabled') ? (
-        <Route path="folder(/:folderId)" component={DriveView}>
-          <Route path="file/:fileId" component={FilesViewer} />
-        </Route>
-      ) : (
-        <Route component={FileExplorer}>
-          <Route path="folder" component={Folder}>
-            <Route path=":folderId">
-              <Route path="file/:fileId" component={FilesViewer} />
-              <Route path="file/:fileId/revision" component={FileHistory} />
-            </Route>
-            {/* Those 2 following routes are needed for the root directory since the url is only /folder, so 
-          next url will be /folder/file/:fileId/ */}
+
+      <Route
+        component={!flag('drive.client-migration.enabled') && FileExplorer}
```

Comment:
> What happens if a Route has not component ?

Not much really, the router will match the children routes but not render a specific component at that level.

> Would it work this way ? […]

I don't think so, but I think I can rewrite it to use the flag only once, hopefully it will be clearer.